### PR TITLE
Add PrometheusAgent to Design docs

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

Hopping that it lands on the website :)

https://prometheus-operator.dev/docs/operator/design/

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: Documentation/design.md [hunk 1]
```diff
@@ -145,3 +146,11 @@ spec:
     webhookConfigs:
     - url: 'http://example.com/'
 ```
+
+## PrometheusAgent
+
+The `PrometheusAgent` custom resource definition (CRD) declaratively defines a desired [Prometheus](https://prometheus.io/docs/prometheus) setup to run in a Kubernetes cluster, but it will necessarily run in [Agent mode](https://prometheus.io/blog/2021/11/16/agent/).
```
Comment (JoaoBraveCoding) [inline Documentation/design.md:152]:
```suggestion
The `PrometheusAgent` custom resource definition (CRD) declaratively defines a desired [Prometheus Agent](https://prometheus.io/blog/2021/11/16/agent/) setup to run in a Kubernetes cluster.
```

File: Documentation/design.md [hunk 2]
```diff
@@ -145,3 +146,11 @@ spec:
...
+
+Similar to Prometheus Server and Prometheus Agent binaries, the `PrometheusAgent` and `Prometheus` CRs are also similar, where the agent is heavily optimized for the remote-write use-case. Both the Agent binary and the CR have several configuration options disabled, e.g. alerting, PrometheusRules selectors, remote-read, storage and thanos sidecars.
```
Comment (JoaoBraveCoding) [inline Documentation/design.md:154]:
```suggestion
Similar to the binaries of Prometheus Server and Prometheus Agent, the `Prometheus` and `PrometheusAgent` CRs are also similar. Inspired in the Agent binary, the Agent CR has several configuration options redacted when compared with regular Prometheus CR, e.g. alerting, PrometheusRules selectors, remote-read, storage and thanos sidecars.
```

**Review by JoaoBraveCoding — COMMENTED** [review]
Great job tidying all the loose ends! Again some possible suggestions feel free to only use them if you feel like it makes the reader's job easier

**Comment by ArthurSens** [comment]
> ng all the loose ends! Again some possible suggestions feel free to only use them if you feel like it makes the reader's job easier

I like it :)

All pushed!

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**